src/test/app/NFToken_test.cpp

```diff
@@ -5041,6 +5137,757 @@ class NFToken_test : public beast::unit_test::suite
         }
     }
 
+    void
+    testIOUWithTransferFee(FeatureBitset features)
+    {
+        using namespace test::jtx;
+
+        testcase("Payments with IOU transfer fees");
+
+        for (auto const& tweakedFeatures :
+             {features - fixUnburnableNFToken, features | fixUnburnableNFToken})
```

Comment:
Not too sure about this, haven't looked too deeply. but just curious if we could avoid having the for-loop by adding 
`testWithFeats(all - fixUnburnableNFToken);`
in `run()`

And then replace `tweakedFeatures[fixUnburnableNFToken]` with `features[fixUnburnableNFToken]` to check if amendment is enabled or not. 

I **think** it reduces number of test being ran? That's how I did it with my remint test cases. Just curious if this approach is applicable. 

CC @scottschurr

Reply:
Actually it may increase the test runs.

I'm going to include the remint change into the amendment as well. And I added `testWithFeats(all - fixUnburnableNFToken);` into my code.

Reply:
There is, as always, a tension between number of test runs and completeness.

Sometimes unit tests do what @ledhed2222 has done here.  That's based on the test-writer's assessment that the amendment they are adding will not have far-reaching changes.  So changes in amendment state are tested locally in just a few places.

Other times it's not so obvious just how localized the changes are.  In that case it is common to run a bunch of the unit tests in the file with different amendments enabled.  Here's an example: https://github.com/XRPLF/rippled/blob/develop/src/test/app/Offer_test.cpp#L5165-L5181

What I see going on is the intent to pile a _bunch_ of changes -- multiple pull requests -- into a single proposed amendment.  That suggests to me that the later approach (like Offer_test.cpp) is probably appropriate.  Any single change due to the amendment may be simple.  But there are many changes and they may interact.

Just my opinion.

Reply:
i'm going to mark this as resolved for now and not change. i think there is an opportunity to simplify the number of test runs happening in this file, but it was already pretty damn complex before and i didn't want to mess with it here